geometry/graham_scan_functions.hpp

```diff
@@ -0,0 +1,232 @@
+/******************************************************************************
+ * @file
+ * @brief Implementation of the [Convex
+ *Hull](https://en.wikipedia.org/wiki/Convex_hull) implementation using [Graham
+ *Scan](https://en.wikipedia.org/wiki/Graham_scan)
+ * @details
+ * In geometry, the convex hull or convex envelope or convex closure of a shape
+ * is the smallest convex set that contains it. The convex hull may be defined
+ * either as the intersection of all convex sets containing a given subset of a
+ * Euclidean space, or equivalently as the set of all convex combinations of
+ * points in the subset. For a bounded subset of the plane, the convex hull may
+ * be visualized as the shape enclosed by a rubber band stretched around the
+ *subset.
+ *
+ * The worst case time complexity of Jarvis’s Algorithm is O(n^2). Using
+ * Graham’s scan algorithm, we can find Convex Hull in O(nLogn) time.
+ *
+ * ### Implementation
+ *
+ * Sort points
+ * We first find the bottom-most point. The idea is to pre-process
+ * points be sorting them with respect to the bottom-most point. Once the points
+ * are sorted, they form a simple closed path.
+ * The sorting criteria is to use the orientation to compare angles without
+ *actually computing them (See the compare() function below) because computation
+ *of actual angles would be inefficient since trigonometric functions are not
+ *simple to evaluate.
+ *
+ * Accept or Reject Points
+ * Once we have the closed path, the next step is to traverse the path and
+ * remove concave points on this path using orientation. The first two points in
+ * sorted array are always part of Convex Hull. For remaining points, we keep
+ *track of recent three points, and find the angle formed by them. Let the three
+ *points be prev(p), curr(c) and next(n). If orientation of these points
+ *(considering them in same order) is not counterclockwise, we discard c,
+ *otherwise we keep it.
```

Comment:
```suggestion
 * track of recent three points, and find the angle formed by them. Let the three
 * points be prev(p), curr(c) and next(n). If the orientation of these points
 * (considering them in the same order) is not counterclockwise, we discard c,
 * otherwise we keep it.
```